libasynql/src/poggit/libasynql/mysqli/MysqliThread.php

```diff
@@ -82,13 +94,20 @@ protected function executeQuery($mysqli, int $mode, string $query, array $params
 		assert($mysqli instanceof mysqli);
 		/** @var MysqlCredentials $cred */
 		$cred = unserialize($this->credentials);
-		while(!$mysqli->ping()){
-			$cred->reconnectMysqli($mysqli);
-			if($this->connError === null){
-				break;
-			}
+		/** @var AttachableThreadedLogger $logger */
+		$logger = unserialize($this->logger);
+		if(!$mysqli->ping()){
+			do{
+				$seconds = min(2 ** $this->attempts++, PHP_INT_MAX);
```

Comment:
This code is confusing. The use of `$x++` here is not idiomatic enough. Please split `++` into a separate statement.

Reply:
Btw, capping by `PHP_INT_MAX` can never happen, because in the 31st sleep we need to sleep for 68 years :upside_down_face: But it's OK to leave it here.